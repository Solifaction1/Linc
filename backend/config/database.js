const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
