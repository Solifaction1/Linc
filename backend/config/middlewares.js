module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
