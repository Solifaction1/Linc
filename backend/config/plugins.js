module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
