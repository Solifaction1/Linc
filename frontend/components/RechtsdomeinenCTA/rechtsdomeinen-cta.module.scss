@import 'styles/vars';
@import 'styles/utilities';

.rechtsdomeinenCTA {
	@extend %page-container;
	position: relative;
	z-index: 3;
	grid-template-areas: 'rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen';
	grid-gap: 0;

	width: 100%;
	margin-right: 0;
	margin-left: 0;
	padding: 0;

	@media (max-width: $break-lg) {
		grid-template-areas:
			'imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile imageMobile'
			'rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer rechtsdomeinenCTAContainer'
			'rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen rechtsDomeinen';
		margin-top: calc(#{$spacing-base} * 55);
	}

	@media (max-width: $break-md) {
		margin-top: calc(#{$spacing-base} * 42.5);
	}

	.imageMobile {
		position: absolute;
		top: -14.5rem;
		left: 0;
		grid-area: imageMobile;

		width: 50vw;
		height: calc(100% + 20.9rem);

		@media (max-width: $break-lg) {
			position: relative;
			top: -10.3rem;
			display: block;

			width: 100%;
			max-width: calc(100% - 7.2rem);
			height: 25rem;
		}

		@media (max-width: $break-md) {
			top: 20rem;
			height: 35rem;
		}

		@media (max-width: $break-s) {
			max-width: calc(100% - 6rem);
		}

		img {
			position: relative;
			object-fit: cover;

			width: 100%;
			max-width: 72rem;
			height: 100%;

			object-position: left;

			@media (max-width: $break-lg) {
				width: calc(100% + calc(#{$spacing-base} * 4.5));
				max-width: 100rem;
				height: auto;
				max-height: 45rem;
				margin-top: -30rem;

				object-position: top;
			}

			@media (max-width: $break-md) {
				top: -20.3rem;
				height: 100%;
				max-height: 35rem;
			}
		}
	}

	.textMobile {
		position: relative;
		grid-area: rechtsdomeinenCTAContainer;

		@media (max-width: $break-lg) {
			margin-top: -21rem;
			padding: 7.5rem 0;
		}

		@media (max-width: $break-md) {
			margin-top: -30.5rem;
		}

		.text p {
			@extend %body-copy-1;
			margin-top: calc(#{$spacing-base} * 4);
		}
	}

	.rechtsdomeinenCTAContainer {
		display: grid;
		grid-area: rechtsdomeinenCTAContainer;
		background-color: $color-green-dark;

		@media (max-width: $break-lg) {
			display: block;
		}

		&::after {
			content: '';
			position: absolute;
			z-index: 2;

			background: $color-green;

			opacity: 0.9;

			mix-blend-mode: multiply;

			@media (max-width: $break-lg) {
				top: 6.4rem;
				right: $page-bleed;

				width: 4.6rem;
				height: 4.6rem;
			}
		}

		.text {
			display: flex;
			flex-direction: column;
			gap: calc(#{$spacing-base} * 5);
			justify-content: center;

			max-width: 57.15rem;
			margin-right: auto;
			margin-left: auto;
			padding-top: calc(#{$spacing-base} * 8);
			padding-bottom: calc(#{$spacing-base} * 8);

			@media (max-width: $break-xlg) {
				gap: calc(#{$spacing-base} * 3.5);
				padding: calc(#{$spacing-base} * 8) calc(#{$spacing-base} * 12)
					calc(#{$spacing-base} * 8) 3.6rem;
			}

			@media (max-width: $break-lg) {
				display: block;
				max-width: unset;
				padding: 7.5rem 3.6rem;
			}

			@media (max-width: $break-s) {
				padding: 7.5rem 2.4rem;
			}

			.title,
			.description {
				color: $color-green !important;
				font-weight: $font-weight-regular;
			}

			.title {
				@media (max-width: $break-lg) {
					font-size: $font-size-title-md-m;
					line-height: 3.5rem;
				}
			}

			.description,
			.cta p {
				font-size: $font-size-text-xlg;
			}

			.description {
				line-height: 3.15rem;

				@media (max-width: $break-lg) {
					font-size: $font-size-text-lg;
					line-height: 2.45rem;
				}
			}

			.cta {
				p {
					line-height: 2.7rem;
				}
			}
		}

		.image {
			position: relative;
			right: calc(#{$spacing-base} * -0);

			width: 100%;
			margin-top: auto;
			@media (min-width: $break-hd) {
				text-align: right;
			}

			img {
				object-fit: cover;

				width: 100%;
				max-height: 46rem;

				object-position: left;
			}

			&::after {
				content: '';
				position: absolute;
				top: 0;
				right: 0;
				z-index: 2;

				width: 7.57rem;
				height: 7.5rem;

				background: $color-green;

				opacity: 0.9;

				mix-blend-mode: multiply;
			}
		}
	}

	.rechtsDomeinen {
		grid-area: rechtsDomeinen;
		background-color: $color-green;

		.rechtsDomeinenContainer {
			display: flex;
			flex-direction: column;
			justify-content: center;

			max-width: 57.15rem;
			height: 100%;
			margin-right: auto;
			padding: calc(#{$spacing-base} * 8) 0 calc(#{$spacing-base} * 8)
				calc(#{$spacing-base} * 11);

			@media (max-width: $break-xlg) {
				padding: calc(#{$spacing-base} * 8) 3.6rem calc(#{$spacing-base} * 8)
					calc(#{$spacing-base} * 11);
			}

			@media (max-width: $break-lg) {
				max-width: unset;
				padding: 7.5rem 3.6rem;
			}

			@media (max-width: $break-s) {
				padding: 7.5rem 2.4rem;
			}

			.cta {
				display: flex;
				flex-direction: row;
				align-items: center;

				width: 100%;
				border-bottom: 1px solid $color-green-dark;

				&:first-of-type {
					border-top: 1px solid $color-green-dark;
				}

				a {
					width: 100%;
					margin-left: 1.5rem;
					padding-top: calc(#{$spacing-base} * 2);
					padding-bottom: calc(#{$spacing-base} * 2);

					color: $color-green-dark;

					text-decoration: none;
					font-size: $font-size-title-md;
					line-height: 5.8rem;

					@media (max-width: $break-lg) {
						font-size: $font-size-title-xs;
					}
				}

				svg {
					width: 5rem;
					height: 5rem;

					fill: $color-green-dark;

					font-weight: 400;
				}

				&:hover {
					background: $color-green-dark-opacity-25;
				}

				&:active {
					background: $color-green-dark;

					a {
						color: $color-green;
					}

					svg {
						fill: $color-green;
					}
				}
			}
		}
	}
}
