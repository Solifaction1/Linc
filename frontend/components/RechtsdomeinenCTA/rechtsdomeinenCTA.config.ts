import RechtsdomeinenCTAData from '@models/rechtsdomeinenCTA.model'

const defaultProps: RechtsdomeinenCTAData = {
	title: 'title',
	description: 'description',
	cta: {
		id: 0,
		title: 'cta title',
		link: '/link',
		externalLink: false,
		buttonType: 'Button primary',
	},
	image: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				alternativeText: 'alternativeText',
				width: 0,
				height: 0,
				hash: 'hash',
				ext: 'ext',
				mime: 'mime',
				size: 0,
				url: 'https://via.placeholder.com/400x500',
			},
		},
	},
	collectionType: {
		data: [
			{
				id: 1,
				attributes: {
					title: 'Rechtsdomein 1',
					slug: 'slug-1',
					image: {
						data: null
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
					color: 'gray'
				}
			},
			{
				id: 2,
				attributes: {
					title: 'Rechtsdomein 2',
					slug: 'slug-2',
					image: {
						data: null
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
					color: 'gray'
				}	
			}
		]
	}
}

export { defaultProps }
