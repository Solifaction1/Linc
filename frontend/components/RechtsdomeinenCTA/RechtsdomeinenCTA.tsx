import Image from 'next/image'
import React from 'react'
import { useMediaQuery } from 'react-responsive'
import RechtsdomeinenCTAData from '@models/rechtsdomeinenCTA.model'
import SwitchCTA from '@templates/SwitchCTA'
import { Icon } from '@templates/switch-icon.config'
import SwitchIcon from '@templates/SwitchIcon'
import { BREAKPOINTS } from '../../constants/breakpoints'
import { defaultProps } from './rechtsdomeinenCTA.config'

import styles from './rechtsdomeinen-cta.module.scss'

const RechtsdomeinenCTA = (props: RechtsdomeinenCTAData) => {
	const image = props.image?.data?.attributes
	const isTablet = useMediaQuery({ query: `(max-width: ${BREAKPOINTS.lg}px)` })

	return (
		<article className={styles.rechtsdomeinenCTA}>
			<div className={styles.rechtsdomeinenCTAContainer}>
				{image && isTablet && (
					<div className={`${styles.imageMobile}`}>
						<Image
							src={image.url}
							alt={image.name}
							width={image.width}
							height={image.height}
							priority
							loader={() => {
								return `/api/custom-loader?url=${encodeURIComponent(
									image.url
								)}?q=${100}`
							}}
						/>
					</div>
				)}
				<div className={`${styles.text} ${isTablet && styles.textMobile}`}>
					{props.title && <h2 className={styles.title}>{props.title}</h2>}
					{props.description && (
						<p className={styles.description}>{props.description}</p>
					)}
					{props.cta && (
						<div className={styles.cta}>
							<SwitchCTA ctaData={props.cta} color="darkGreen" />
						</div>
					)}
				</div>

				{image && !isTablet && (
					<div className={`${styles.image}`}>
						<Image
							src={image.url}
							alt={image.name}
							width={image.width}
							height={image.height}
							priority
							loader={() => {
								return `/api/custom-loader?url=${encodeURIComponent(
									image.url
								)}?q=${100}`
							}}
						/>
					</div>
				)}
			</div>

			<div className={styles.rechtsDomeinen}>
				<div className={styles.rechtsDomeinenContainer}>
					{props.collectionType?.data.map((rechtsDomein) => (
						<div className={styles.cta} key={rechtsDomein.id}>
							<SwitchCTA
								ctaData={{
									title: rechtsDomein.attributes.title ?? '',
									link: `/onze-rechtsdomeinen#${rechtsDomein.id}`,
									externalLink: false,
									buttonType: 'Link',
								}}
							/>

							<SwitchIcon icon={Icon.ANGLE_RIGHT} />
						</div>
					))}
				</div>
			</div>
		</article>
	)
}

RechtsdomeinenCTA.defaultProps = defaultProps

export default RechtsdomeinenCTA
