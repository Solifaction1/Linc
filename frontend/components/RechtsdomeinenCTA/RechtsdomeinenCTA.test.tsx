import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { defaultProps } from './rechtsdomeinenCTA.config'
import RechtsdomeinenCTA from './RechtsdomeinenCTA'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('RechtsdomeinenCTA', () => {
	it('renders a RechtsdomeinenCTA', () => {
		const { container, getByText } = render(
			<RechtsdomeinenCTA {...defaultProps} />
		)

		expect(container).toMatchSnapshot()

		expect(container.querySelector('article')).toBeVisible()
		expect(
			container.getElementsByClassName('rechtsdomeinenCTA')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('rechtsdomeinenCTAContainer')[0]
		).toBeVisible()

		expect(container.getElementsByClassName('text')[0]).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'h2' })).toBeVisible()

		expect(container.getElementsByClassName('description')[0]).toBeVisible()
		expect(getByText('description', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('cta')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('button false primary md false')[0]
		).toBeVisible()
		expect(getByText('cta title', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('image')[0]).toBeVisible()

		const img = container.querySelector('img')
		expect(img).toBeVisible()
		expect(img?.getAttribute('alt')).toEqual('name')
		expect(img?.getAttribute('width')).toEqual('0')
		expect(img?.getAttribute('height')).toEqual('0')
		expect(img?.getAttribute('style')).toEqual('color: transparent;')
		expect(img?.getAttribute('src')).toEqual(
			'/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100'
		)

		expect(container.getElementsByClassName('rechtsDomeinen')[0]).toBeVisible()

		expect(container.getElementsByClassName('cta')[1]).toBeVisible()
		expect(
			container.querySelector('a[href="/onze-rechtsdomeinen#1"]')
		).toBeVisible()
		expect(
			container.getElementsByClassName('link link md false')[0]
		).toBeVisible()
		expect(getByText('Rechtsdomein 1', { selector: 'a' })).toBeVisible()
		expect(container.querySelectorAll('svg')[0]).toBeVisible()
		expect(container.querySelectorAll('path')[0]).toBeVisible()

		expect(container.getElementsByClassName('cta')[2]).toBeVisible()
		expect(
			container.querySelector('a[href="/onze-rechtsdomeinen#2"]')
		).toBeVisible()
		expect(
			container.getElementsByClassName('link link md false')[1]
		).toBeVisible()
		expect(getByText('Rechtsdomein 2', { selector: 'a' })).toBeVisible()
		expect(container.querySelectorAll('svg')[1]).toBeVisible()
		expect(container.querySelectorAll('path')[1]).toBeVisible()
	})
})
