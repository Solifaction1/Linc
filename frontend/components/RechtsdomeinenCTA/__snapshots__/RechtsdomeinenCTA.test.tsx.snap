// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RechtsdomeinenCTA renders a RechtsdomeinenCTA 1`] = `
<div>
  <article
    class="rechtsdomeinenCTA"
  >
    <div
      class="rechtsdomeinenCTAContainer"
    >
      <div
        class="text false"
      >
        <h2
          class="title"
        >
          title
        </h2>
        <p
          class="description"
        >
          description
        </p>
        <div
          class="cta"
        >
          <button
            class="button false primary md false  darkGreen"
            type="button"
          >
            <p
              class="label"
            >
              cta title
            </p>
          </button>
        </div>
      </div>
      <div
        class="image"
      >
        <img
          alt="name"
          data-nimg="1"
          decoding="async"
          height="0"
          src="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100"
          srcset="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100 1x"
          style="color: transparent;"
          width="0"
        />
      </div>
    </div>
    <div
      class="rechtsDomeinen"
    >
      <div
        class="rechtsDomeinenContainer"
      >
        <div
          class="cta"
        >
          <a
            class="link  link md false"
            href="/onze-rechtsdomeinen#1"
          >
            Rechtsdomein 1
          </a>
          <svg
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
            />
          </svg>
        </div>
        <div
          class="cta"
        >
          <a
            class="link  link md false"
            href="/onze-rechtsdomeinen#2"
          >
            Rechtsdomein 2
          </a>
          <svg
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </article>
</div>
`;
